### PR TITLE
Keep the main save for a pilot at the top of the list, above snapshots

### PR DESCRIPTION
**Pull request by warp-core**
**Feature:** This PR implements the feature request detailed and discussed in issue #9259

## Feature Details
This ensures that the main, non-snapshot save file will always be sorted to the top of the list of files for each pilot.

## Testing Done
Have all save files have the same file system timestamp and see that the non-snapshot save is at the top of the list even though this isn't where it would appear alphabetically.

### CODE REVIEW

File: source/LoadPanel.cpp [hunk 1]
```diff
@@ -450,20 +451,34 @@ void LoadPanel::UpdateLists()
...
+		{
+			swap(savesList.front(), savesList.back());
+			notSnapshotOnly.insert(pilotName);
+		}
 	}
 
 	for(auto &it : files)
-		sort(it.second.begin(), it.second.end(),
+	{
+		// Don't include the first itme in the sort if this pilot has a non-snapshot save.
```
Comment (RisingLeaf) [inline source/LoadPanel.cpp:471]:
```suggestion
		// Don't include the first item in the sort if this pilot has a non-snapshot save.
```

File: source/LoadPanel.cpp [hunk 2]
```diff
@@ -441,6 +441,7 @@ void LoadPanel::UpdateLists()
 	files.clear();
 
 	vector<string> fileList = Files::List(Files::Saves());
+	set<string> notSnapshotOnly;
```
Comment (RisingLeaf) [inline source/LoadPanel.cpp:444]: So these are saves that are not snapshots?
The name is a bit confusing to me.

**Review by RisingLeaf — APPROVED** [review]
Code looks good to me

File: source/LoadPanel.cpp [hunk 3]
```diff
@@ -450,20 +451,34 @@ void LoadPanel::UpdateLists()
...
+			notSnapshotOnly.insert(pilotName);
+		}
 	}
 
 	for(auto &it : files)
-		sort(it.second.begin(), it.second.end(),
+	{
+		// Don't include the first item in the sort if this pilot has a non-snapshot save.
+		auto start = it.second.begin();
+		if(notSnapshotOnly.count(it.first))
```
Comment (quyykk) [inline source/LoadPanel.cpp:473]: You could instead check if the first entry's file name includes a `~`.

File: source/LoadPanel.cpp (hunk 2 above)
Comment (warp-core) [reply source/LoadPanel.cpp]: Using @quyykk's suggestion and removing the `set` anyway.

**Review by quyykk — APPROVED** [review]

**Pull request merged**